XML-IN-KDL.md

```diff
@@ -0,0 +1,79 @@
+
+XML has several types of nodes, corresponding to certain KDL constructs:
+
+* Elements, which have an element name, zero or more attribute, and zero or more children. These are encoded directly as KDL nodes, using the nodename, properties, and children nodes.
+* Raw text. In "pure" XML dialects, where raw text only appears as the sole child of an element (never mixed with other elements as siblings), this is generally encoded as a final string argument in a KDL node; in "mixed" XML dialects, it can be encoded as a special KDL node with the name `-`.
+* Comments. These are encoded as KDL nodes with the name `!`.
+* Processing Instructions. These are encoded similarly to elements if their contents are sufficiently structured, with a `?` in front of their node name. If they're not sufficiently structured, their contents are just strings.
+* Doctypes. These are encoded like unstructured PIs, just with the node name `!doctype`.
+
+----
+
+XML elements and KDL nodes have a direct correspondence. In XiK, an XML element is encoded in KDL by:
+
+* making the element name the KDL node name
+* making the attributes into KDL properties
+* making the child nodes as KDL child nodes
+
+For example, the XML `<element foo="bar"><child baz="qux" /></element>` is encoded into XiK as `element foo="bar" { child baz="qux" }`.
+
+XML namespaces are encoded the same as XML: the node name simply contains a `:` character. Note that KDL identifier syntax allows `:` directly in an ident, so a name like `xml:space` or `xlink:href` is a valid node or property name.
+
+----
+
+Raw text contents of an element can be encoded in two possible ways.
+
+If the element contains *only* text, it should be encoded as a final string unnamed argument. For example, the XML `<a href="http://example.com">here's a link</a>` can be encoded as `a href="http://example.com" "here's a link"`.
+
+If the element contains mixed text and element children, the text can be encoded as a KDL node with the name `-` with a single string unnamed argument. For example, the XML `<span>some <b>bold</b> text</span>` can be encoded as `span { - "some "; b "bold"; - " text" }`.
+
+An element that contains only text *is allowed to* encode it as `-` children. For example, `<span>foo</span>` *may* be encoded as `span { - "foo" }` instead of `span "foo"`. However, an element cannot mix the "final string attribute" with child nodes; `span "foo" { b "bar" }` is an **invalid** encoding of `<span>foo<b>bar</b></span>`. (It must be encoded as `span { - "foo"; b "bar" }`.)
+
+CDATA sections are not preserved in this encoding, as they are merely a source convenience so you don't have to escape a bunch of characters. They are encoded as normal textual contents would be.
+
+-----
+
+Comments are encoded as a special node name `!`, with a single unnamed string argument containing the comment's value (everything between the `<!--` and the `-->`). For example, `<!-- comment! -->` is encoded as the node `! " comment! "`.
```

Comment:
maybe I don't understand XML enough, but what's the reason for using a node for comments, instead of KDL's existing comment facilities?

Reply:
I'm *presuming* that XML comments should round-trip (but KDL comments shouldn't, at least not necessarily).

I could be convinced otherwise.

Reply:
Maybe just have a note to only use this when authoring if you *specifically* want the comment to reflect into the XML document; otherwise stick with KDL comments?

Reply:
I like that idea!

Although I'm also not opposed to saying you can/should use `/* */` comments (only) if you want them to round-trip, and all other KDL comments can be ignored as you said (or even disallowed. `/-` is pretty weird when it comes to comment formats)

Reply:
Okay, done. I said that they're encoded as multiline comments, but left `!` nodes as an alternative option specifically if you're using a KDL toolchain that discards comments and need them reflected into the XML.